c_src/stb_image_nif.c

```diff
@@ -259,6 +259,144 @@ static ERL_NIF_TERM to_file(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[])
     return enif_make_atom(env, "ok");
 }
 
+typedef struct WriteChunk {
+    struct WriteChunk *next;
+    void *data;
+    size_t size;
+} WriteChunk;
+
+typedef struct {
+    WriteChunk *head;
+    WriteChunk *last;
+    size_t size;
+    bool out_of_memory;
+} WriteContext;
+
+static void write_chunk(void *context_, void *data, int size) {
+    WriteContext *context = (WriteContext *) context_;
+
+    if (context->out_of_memory) {
+        return;
+    }
+
+    WriteChunk *chunk = enif_alloc(sizeof(WriteChunk));
+    void *chunk_data = enif_alloc(size);
+
+    if (chunk == NULL || chunk_data == NULL) {
+        free(chunk);
+        free(chunk_data);
+        context->out_of_memory = true;
+        return;
+    }
+
+    memcpy(chunk_data, data, size);
+
+    chunk->next = NULL;
+    chunk->data = chunk_data;
+    chunk->size = size;
+
+    if (context->head == NULL) {
+        context->head = context->last = chunk;
+    } else {
+        context->last->next = chunk;
+        context->last = chunk;
+    }
+
+    context->size += size;
+}
+
+static void finalize_write(WriteContext *context, ErlNifEnv *env, ERL_NIF_TERM *binary) {
+    if (!context->out_of_memory) {
+        void *buffer = enif_make_new_binary(env, context->size, binary);
+
+        if (buffer == NULL) {
+            context->out_of_memory = true;
+        } else {
+            for (WriteChunk *chunk = context->head; chunk != NULL; chunk = chunk->next) {
+                memcpy(buffer, chunk->data, chunk->size);
+                buffer += chunk->size;
+            }
+        }
+    }
+
+    WriteChunk *chunk = context->head;
+    WriteChunk *next = NULL;
+
+    while (chunk != NULL) {
+        next = chunk->next;
+        enif_free(chunk->data);
+        enif_free(chunk);
+        chunk = next;
+    }
+}
+
+static ERL_NIF_TERM to_binary(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
+    if (argc != 5) {
+        return error(env, "expecting 5 arguments: extension, data, height, width, and number of channels");
```

Comment:
```suggestion
        return error(env, "expecting 5 arguments: format, data, height, width, and number of channels");
```

?

Reply:
Maybe change the remaining vars from `extension` too.

Reply:
Yeah, thanks, I missed those.